src/test/java/com/example/Restaurant_Management_System_REST_API/service/RestaurantOrderServiceTest.java

```diff
@@ -30,55 +31,72 @@ class RestaurantOrderServiceTest {
         tableRepository = mock(TableRepository.class);
-        tableService = mock(TableService.class);
-        restaurantOrderService = new RestaurantOrderService(restaurantOrderRepository, modelMapper, tableService);
-        restaurantOrderDTO = mock(RestaurantOrderDTO.class);
+        TableService tableService = mock(TableService.class);
+        MenuRecordRepository menuRecordRepository = mock(MenuRecordRepository.class);
+        MenuRecordService menuRecordService = new MenuRecordService(menuRecordRepository, modelMapper);
+        InventoryItemService inventoryItemService = mock(InventoryItemService.class);
+        RestaurantOrderMenuRecordService restaurantOrderMenuRecordService = mock(RestaurantOrderMenuRecordService.class);
+        restaurantOrderService = new RestaurantOrderService(restaurantOrderRepository, modelMapper, tableService,
+                menuRecordService, inventoryItemService, restaurantOrderMenuRecordService);
+        restaurantOrderResponseDTO = mock(RestaurantOrderResponseDTO.class);
+        restaurantOrderRequestDTO = mock(RestaurantOrderRequestDTO.class);
         restaurantOrder = mock(RestaurantOrder.class);
         id = 1L;
     }
 
+    @Test
+    public void create_ShouldThrowNotFoundInDatabaseException_WhenMenuRecordsIsNull() {
+        // Arrange
+        RestaurantOrderRequestDTO restaurantOrderDTO = new RestaurantOrderRequestDTO();
+        restaurantOrderDTO.setMenuRecords(null);
+
+        // Act and Assert
+        assertThrows(NotFoundInDatabaseException.class, () -> restaurantOrderService.create(restaurantOrderDTO));
+    }
 
     @Test
-    public void create_ShouldInteractWithDependenciesCorrectlyAndReturnRestaurantOrderDTO_WhenRestaurantOrderDTOIsGiven()
-            throws CustomerAlreadyHasReservationException, NotFoundInDatabaseException {
-        //Arrange
-        when(modelMapper.map(restaurantOrderDTO, RestaurantOrder.class)).thenReturn(restaurantOrder);
-        when(restaurantOrderRepository.save(restaurantOrder)).thenReturn(restaurantOrder);
-        when(modelMapper.map(restaurantOrder, RestaurantOrderDTO.class)).thenReturn(restaurantOrderDTO);
+    public void create_ShouldThrowNotFoundInDatabaseException_WhenMealNotInMenu() {
+        // Arrange
+        RestaurantOrderRequestDTO restaurantOrderDTO = createRestaurantOrderDTOWithInvalidMeal();
 
-        //Act
-        RestaurantOrderDTO actual = restaurantOrderService.create(restaurantOrderDTO);
+        // Act and Assert
+        assertThrows(NotFoundInDatabaseException.class, () -> restaurantOrderService.create(restaurantOrderDTO));
+    }
 
-        //Assert
-        assertEquals(restaurantOrderDTO, actual);
+    private RestaurantOrderRequestDTO createRestaurantOrderDTOWithInvalidMeal() {
+        RestaurantOrderRequestDTO restaurantOrderDTO = new RestaurantOrderRequestDTO();
+        restaurantOrderDTO.setMenuRecords(Arrays.asList(new MenuRecordForOrderDTO(1L, "Not valid meal", 1.0)));
+        return restaurantOrderDTO;
```

Comment:
There is a room to improve by avoiding magic numbers.
An explanation code regarding this matter could look like this:
```
private static final Long INVALID_MEAL_ID = 1L;
private static final String INVALID_MEAL_NAME = "Not valid meal";
private static final Double INVALID_MEAL_PRICE = 1.0;

private RestaurantOrderRequestDTO createRestaurantOrderDTOWithInvalidMeal() {
    RestaurantOrderRequestDTO restaurantOrderDTO = new RestaurantOrderRequestDTO();
    restaurantOrderDTO.setMenuRecords(Arrays.asList(new MenuRecordForOrderDTO(INVALID_MEAL_ID, INVALID_MEAL_NAME, INVALID_MEAL_PRICE)));
    return restaurantOrderDTO;
}
```